hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestCommandQueueReportHandler.java

```diff
@@ -77,35 +79,51 @@ public void resetEventCollector() throws IOException {
   public void testQueueReportProcessed() throws NodeNotFoundException {
     DatanodeDetails dn = MockDatanodeDetails.randomDatanodeDetails();
     nodeManager.register(dn, null, null);
+
+    // Add some queued commands to be sent to the DN on this heartbeat. This
+    // means the real queued count will be the value reported plus the commands
+    // sent to the DN.
+    Map<SCMCommandProto.Type, Integer> commandsToBeSent = new HashMap<>();
+    commandsToBeSent.put(SCMCommandProto.Type.valueOf(1), 2);
+    commandsToBeSent.put(SCMCommandProto.Type.valueOf(2), 1);
+
     SCMDatanodeHeartbeatDispatcher.CommandQueueReportFromDatanode
-        commandReport = getQueueReport(dn);
+        commandReport = getQueueReport(dn, commandsToBeSent);
     commandQueueReportHandler.onMessage(commandReport, this);
 
     int commandCount = commandReport.getReport().getCommandCount();
     for (int i = 0; i < commandCount; i++) {
       int storedCount = nodeManager.getNodeQueuedCommandCount(dn,
           commandReport.getReport().getCommand(i));
-      Assertions.assertEquals(commandReport.getReport().getCount(i),
-          storedCount);
+      int expectedCount = commandReport.getReport().getCount(i);
+      // For the first two commands, we added extra commands
```

Comment:
Hello Stephen! Just for my self-learning, why the first two commands we add extra commands~? Thanks~ :pray:

Reply:
This is just to test the new part of the code - that the queued commands in the list are added to the totals in the report count. I did not add extra commands to all of them, so I could see the others work ok when there isn't extra commands.

Reply:
got it! thanks Stephen!